internal/pkg/runtime/engines/singularity/container_linux.go

```diff
@@ -571,10 +570,29 @@ func (c *container) mountImage(mnt *mount.Point) error {
 	}
 
 	path := fmt.Sprintf("/dev/loop%d", number)
-	sylog.Debugf("Mounting loop device %s to %s\n", path, mnt.Destination)
-	_, err = c.rpcOps.Mount(path, mnt.Destination, mnt.Type, flags, optsString)
+
+	sylog.Debugf("Mounting loop device %s to %s of type %s\n", path, mnt.Destination, mnt.Type)
+
+	mountType := mnt.Type
+
+	if mnt.Type == "encryptfs" {
+		cryptDev, err := c.rpcOps.Decrypt(offset, path)
+
+		if err != nil {
+			return fmt.Errorf("Unable to decrypt the file system: %s", err)
```

Comment:
Its better to not use caps in error messages

```suggestion
			return fmt.Errorf("unable to decrypt the file system: %s", err)
```